### DISCUSSION TITLE
How do you pass additional args to the optimistic update functions?

### DISCUSSION THREAD

**Discussion by taylorcantwell** [urql-graphql/urql#2106] [Q&A] [answered]
Hopefully someone could point out if I'm doing something wrong here. Basically the optimisticVariables are not appearing in the function, only the variables specified in the mutation. Referencing: https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#variables-for-optimistic-updates

```
cacheExchange({
         optimistic: {
            mutationName: (variables, cache, info) => {....} ......

   mutation({
            optimisticVariables: {abc: "123"},
            variableSpecifiedInMutation: {
            ....
            },
         });

```

**Comment by kitten** [comment, marked as answer]
I think the docs aren't precise enough here and I'll have to fix them up. Apologies! 🙏

Basically, variables and arguments are different here. You define your variables, e.g. `$var` and your field args, e.g. `input: $var`. Your field will only see `input: ...` since it takes a subset of the variables. GraphQL basically allows Graphcache to also filter variables before they're sent to the API.

That means that during optimistic mutations however, were able to send more variables because they haven't been filtered for the API yet. Hence you'll see more variables that you've passed, but not more arguments. The variables will still be filtered later however 

To then access the raw variables, you can use `ctx.variables` on any optimistic mutation function, which isn't the arguments object. That `ctx` is basically what you've got as `info` in your code snippet.

This is an escape hatch because it's obviously a raw object and it will happen to be exactly what you've passed, but not safely filtered.

**Reply by taylorcantwell** [reply]
Thanks for the help, Kitten.
